### PR TITLE
pulseaudio 17.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **pulseaudio/17.0.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/1)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/2)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/3)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/4)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/5)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/8)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/9)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/10)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/attempts/12)

**Review by RubenRBS — COMMENTED** [review]
Can Ii get some references as to when this went from beinig a `libbrbary` wiith `shared` option to only being a `shared-liibrbary` exclusively? Thanks!

**Comment by ericLemanissier** [comment]
friendly ping @RubenRBS

**Review by valgur — APPROVED** [review]

**Comment by uilianries** [comment]
What's the advantage of using Meson only now? As far as I see, the previous recipe using Autotools is just simple like this one. Both Autotools and Meson are much complicated than CMake, for instance.

**Comment by valgur** [comment]
@uilianries v15+ has dropped the autotools-based build system entirely. Also, autotools is a major pain to set up on Windows.

**Comment by ericLemanissier** [comment]
https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/commit/fc40f046dd9f2b27953af78025c7416f19f9c2e2#9f621eb5fd3bcb2fa5c7bd228c9b1ad42edc46c8_1_30

**Comment by uilianries** [comment]
> [gitlab.freedesktop.org/pulseaudio/pulseaudio/-/commit/fc40f046dd9f2b27953af78025c7416f19f9c2e2#9f621eb5fd3bcb2fa5c7bd228c9b1ad42edc46c8_1_30](https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/commit/fc40f046dd9f2b27953af78025c7416f19f9c2e2#9f621eb5fd3bcb2fa5c7bd228c9b1ad42edc46c8_1_30)

@ericLemanissier Please, try to keep this kind of information in the description. This PR is not just a bump version.

File: recipes/pulseaudio/meson/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        if self.options.get_safe("with_x11"):
+            self.requires("xorg/system")
+        if self.options.with_openssl:
+            self.requires("openssl/[>=1.1 <4]")
+        if self.options.with_dbus:
+            self.requires("dbus/1.15.8")
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("pulseaudio supports only linux currently")
```
Comment (uilianries) [inline recipes/pulseaudio/meson/conanfile.py:69]:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} recipe is only compatible with Linux right now. Contributions are welcome.")
```

The changelog says is possible to build on Windows now.

File: recipes/pulseaudio/meson/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+from conan.tools.layout import basic_layout
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class PulseAudioConan(ConanFile):
+    name = "pulseaudio"
+    description = "PulseAudio is a sound system for POSIX OSes, meaning that it is a proxy for sound applications."
+    topics = ("sound",)
```
Comment (uilianries) [inline recipes/pulseaudio/meson/conanfile.py:17]:
```suggestion
    topics = ("sound", "audio", "sound-server")
```

File: recipes/pulseaudio/meson/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/pulseaudio/meson/conanfile.py:69]: I got this error `..\src\meson.build:63:0: ERROR: Assert failed: Prefix is not absolute: "D:/a/_temp/.c2/p/b/pulse0b1765b681bfa/p"` https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/job/20666352739#step:7:306

As the package does not support windows currently https://github.com/conan-io/conan-center-index/blob/master/recipes/pulseaudio/all/conanfile.py#L78-L79 I chose not to loose time on this

Comment (uilianries) [reply recipes/pulseaudio/meson/conanfile.py]: No problem at all, I'm sure trying to support Windows will be a challenge. It can be done later for sure.

**Review by uilianries — APPROVED** [review]
LGTM. Thank you for porting to meson.

File: recipes/pulseaudio/meson/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            if not self.dependencies["fftw"].options.precision_single:
+                raise ConanInvalidConfiguration(
+                     "Pulse audio uses fftw single precision. "
+                     "Either set option -o fftw/*:precision_single=True or -o pulseaudio/*:with_fftw=False"
+                )
+
+    def build_requirements(self):
+        self.tool_requires("meson/1.3.1")
+        if not self.conf.get("tools.gnu:pkg_config", check_type=str):
+            self.tool_requires("pkgconf/2.0.3")
```
Comment (jwillikers) [inline recipes/pulseaudio/meson/conanfile.py:81]:
```suggestion
            self.tool_requires("pkgconf/2.1.0")
```

File: recipes/pulseaudio/meson/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+        if self.options.get_safe("with_fftw"):
+            if not self.dependencies["fftw"].options.precision_single:
+                raise ConanInvalidConfiguration(
+                     "Pulse audio uses fftw single precision. "
+                     "Either set option -o fftw/*:precision_single=True or -o pulseaudio/*:with_fftw=False"
+                )
+
+    def build_requirements(self):
+        self.tool_requires("meson/1.3.1")
```
Comment (jwillikers) [inline recipes/pulseaudio/meson/conanfile.py:79]:
```suggestion
        self.tool_requires("meson/1.3.2")
```

**Review by jwillikers — APPROVED** [review]

**Pull request merged**